### ISSUE TITLE
Phantom cars

### ISSUE THREAD

**Issue by insx72** [djyt/cannonball#148]
"Phantom" cars appear from nowhere for a split second when the stage changes. They are solid as I got knocked off the road by one!

**Comment by henhouseharry** [comment]
I've noticed this too. Could it be they are from the other road?

**Comment by J1mbo** [comment]
I didn't know they could be crashed into! IIRC it's a 60fps or hires bug.

**Comment by J1mbo** [comment]
> I've noticed this too. Could it be they are from the other road?

Yes this is exactly what is happening. In 60fps mode, objects are spawned every frame but only culled at 30 (ie every other frame), so I think this is the issue. Not got a fix yet.

**Comment by J1mbo** [comment]
Actually what I wrote above is not correct. The issue is that when the game goes from single road back to dual road, the sprites are drawn for one frame with out-of-date road info.

It's kind of difficult to fix this without altering the gameplay. But, I've put a fix for this in my fork which seems to work. Essentially it marks the problem sprites (any that are on the hidden road during the single-road stage exit) then suppresses display of those for two frames when we go back to dual-road.

Enjoy!
